### PR TITLE
write existing rules as per new design.

### PR DESCRIPTION
**Pull request by thiyagu06**
Follow up of #400 .

### CODE REVIEW

File: lib/build.gradle [hunk 1]
```diff
@@ -8,6 +8,7 @@ dependencies {
     testImplementation 'org.slf4j:slf4j-jdk14:2.0.3'
     testImplementation 'org.openapitools.empoa:empoa-simple-models-impl:2.1.0'
     testImplementation 'nl.jqno.equalsverifier:equalsverifier:3.10.1'
+    implementation group: 'org.apache.commons', name: 'commons-lang3', version: '3.12.0'
```
Comment (JFCote) [inline lib/build.gradle:11]: Please keep the implementation near the other implementation. I like to keep `implementation` and `testImplementation` in separated blocks :)

File: lib/src/test/java/org/openapitools/openapistylevalidator/rules/ContactInfoRuleTest.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                .info(OASFactory.createInfo()
+                        .license(OASFactory.createLicense())
+                        .contact(OASFactory.createContact()));
+        List<StyleError> styleErrors = rule.execute(openapi);
+        assertEquals(1, styleErrors.size());
+        assertEquals("*ERROR* Section: APIInfo: 'contact' -> At least one field should be present and not empty. name|url|email", styleErrors.get(0)
+                .toString());
+    }
+
+    private OpenAPI createOpenApi() {
```
Comment (JFCote) [inline lib/src/test/java/org/openapitools/openapistylevalidator/rules/ContactInfoRuleTest.java:48]: I would rename this `createValidOpenApi`

File: lib/src/test/java/org/openapitools/openapistylevalidator/rules/InfoDescriptionRuleTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Test
+    public void invalidOpenAPI() {
+        OpenAPI openapi = OASFactory.createOpenAPI().openapi("3.0.1").info(OASFactory.createInfo().description(""));
+        List<StyleError> styleErrors = rule.execute(openapi);
+        assertEquals(1, styleErrors.size());
+        assertEquals("*ERROR* Section: APIInfo: 'description' -> Should be present and not empty", styleErrors.get(0).toString());
+    }
+
+    private OpenAPI createOpenApi() {
```
Comment (JFCote) [inline lib/src/test/java/org/openapitools/openapistylevalidator/rules/InfoDescriptionRuleTest.java:33]: I would rename this `createValidOpenApi`

File: lib/src/test/java/org/openapitools/openapistylevalidator/rules/LicenseRuleTest.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                .openapi("3.0.1")
+                .info(OASFactory.createInfo()
+                        .license(OASFactory.createLicense()));
+        List<StyleError> styleErrors = rule.execute(openapi);
+        assertEquals(1, styleErrors.size());
+        assertEquals("*ERROR* Section: APIInfo: 'name|url' in license -> At least one field should be present and not empty", styleErrors.get(0)
+                .toString());
+    }
+
+    private OpenAPI createOpenApi() {
```
Comment (JFCote) [inline lib/src/test/java/org/openapitools/openapistylevalidator/rules/LicenseRuleTest.java:46]: I would rename this `createValidOpenApi`

File: lib/src/test/java/org/openapitools/openapistylevalidator/rules/OpenAPIRuleTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    @Test
+    public void invalidOpenAPI() {
+        OpenAPI openapi = OASFactory.createOpenAPI().openapi("3.0.1");
+        List<StyleError> styleErrors = rule.execute(openapi);
+        assertEquals(1, styleErrors.size());
+        assertEquals("*ERROR* Section: OpenAPI: 'paths,components' -> Should have at least one of paths or components", styleErrors.get(0).toString());
+    }
+
+    private OpenAPI createOpenApi() {
```
Comment (JFCote) [inline lib/src/test/java/org/openapitools/openapistylevalidator/rules/OpenAPIRuleTest.java:31]: I would rename this `createValidOpenApi`

**Review by JFCote — CHANGES_REQUESTED** [review]
Very nice first draft! Here are a couple of small tweak I suggest. I can also see there seems to be a problem with the automatic build.

**Comment by thiyagu06** [comment]
@JFCote I moved few existing rules https://github.com/OpenAPITools/openapi-style-validator/blob/master/lib/src/main/java/org/openapitools/openapistylevalidator/OpenApiSpecStyleValidator.java#L32-L34 into the new design. I feel we can take another one or two PR to move the remaining PR's. any thoughts?

**Comment by JFCote** [comment]
> @JFCote I moved few existing rules https://github.com/OpenAPITools/openapi-style-validator/blob/master/lib/src/main/java/org/openapitools/openapistylevalidator/OpenApiSpecStyleValidator.java#L32-L34 into the new design. I feel we can take another one or two PR to move the remaining rules. any thoughts?

Hey @thiyagu06 , just came back from vacation. Will take a look at what you have done in the coming week! But yeah I think we can move a couple other rules but don't create a PR that is too big please :)

**Review by JFCote — APPROVED** [review]
LGTM. I suggest we merge this right now and you create a new PR for other rules

**Comment by JFCote** [comment]
@thiyagu06 Merged! Please create another branch and PR for the other rules :)

**Pull request merged**